src/sqlfluff/rules/L043.py

```diff
@@ -128,78 +126,77 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             and context.segment.segments[0].name == "case"
         ):
             # We can't reduce CASE expressions with multiple WHEN clauses.
-            if (
-                len(
-                    [
-                        segment
-                        for segment in context.segment.segments
-                        if segment.name == "when"
-                    ]
-                )
-                > 1
-            ):
+            when_clauses = [
+                segment
+                for segment in context.segment.segments
+                if segment.is_type("when_clause")
+            ]
+            if len(when_clauses) > 1:
                 return None
 
-            # Find condition expression.
```

Comment:
Here replaced a chain of 3 `while` statements that share the same index with a `for` loop and a list comprehension. (I don't recall the details, but I think some Python linters now complain on *any* `while` statement because they're trickier than `for` loops.)